### PR TITLE
To solve problem of timer apocalypse, changing type of dueMap to Map

### LINKED ISSUE #2615: Fix time apocalypse in timer.ts
In js/timers.ts, next TODO is told.
```typescript
// We'll subtract EPOCH every time we retrieve the time with Date.now(). This
// ensures that absolute time values stay below UINT32_MAX - 2, which is the
// maximum object key that EcmaScript considers "numerical". After running for
// about a month, this is no longer true, and Deno explodes.
// TODO(piscisaureus): fix that ^.
```

So fix this. 😃

### PR DESCRIPTION
**Pull request by 7k8m**
This will fix #2615

### CODE REVIEW

**Comment by 7k8m** [comment]
> [CLA assistant check](https://cla-assistant.io/denoland/deno?pullRequest=2616)
> Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/denoland/deno?pullRequest=2616) before we can accept your contribution.
> 
> **Tomohito Nakayama** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/denoland/deno?pullRequest=2616) it.

**Comment by 7k8m** [comment]
Sorry, mistakenly closed. Reopened again. 🙏

File: js/timers.ts [hunk 1]
```diff
@@ -14,28 +14,34 @@ interface Timer {
...
 
 let globalTimeoutDue: number | null = null;
 
 let nextTimerId = 1;
 const idMap = new Map<number, Timer>();
-const dueMap: { [due: number]: Timer[] } = Object.create(null);
+const dueMap = new Map<number, Timer[]>();
+
+function addToDueMap(due: number, timer: Timer): Timer[] {
+  var list = dueMap.get(due);
```
Comment (zekth) [inline js/timers.ts]: use `let`

File: js/timers.ts [hunk 2]
```diff
@@ -14,28 +14,34 @@ interface Timer {
...
+  if (list === undefined) {
```
Comment (zekth) [inline js/timers.ts]: why not using `if(!list)` ?

**Comment by 7k8m** [comment]
👍

**Comment by 7k8m** [comment]
😕

**Comment by piscisaureus** [comment]
This makes adding/removing as well as firing a timer an O(n) operation where n == the number of active timers.
I'm not loving it. @ry, wdyt?

**Comment by ry** [comment]
That doesn't sounds ideal.

@7k8m Is this an actual bug that you ran into?

**Comment by hayd** [comment]
Is an alternative, if you need to insert a timer for `T` >= `APOCALYPSE`, to insert a thunk for `APOCALYPSE - 1` which itself adds another timer for `T - APOCALYPSE + 1` etc?

_It is unclear the usecase for such long timers..._

**Comment by 7k8m** [comment]
> That doesn't sounds ideal.
> 
> @7k8m Is this an actual bug that you ran into?

Exactly, not actual bug which I am urged to fix.
I think it might be one answer to do nothing until any issue is raised from exact use case. 🙄

**Comment by ry** [comment]
@7k8m Cool, that's what I'm thinking. I'm closing this but it'll be in the history in case we run into this again. Thanks for the patch!

**Comment by hayd** [comment]
...

**Pull request closed without merging**